tests/e2e/test_api.py

```diff
@@ -1120,7 +1120,9 @@ def get_last_log_line(logs: str) -> str:
         timeout=BASIC_ENDPOINTS_TIMEOUT,
     )
     assert response.status_code == requests.codes.ok
-    time.sleep(OLS_USER_DATA_COLLECTION_INTERVAL + 1)
+    # ensure the script have enought time to send the payload before
+    # we pull its logs
+    time.sleep(OLS_USER_DATA_COLLECTION_INTERVAL + 5)
```

Comment:
@onmete if you find us hitting this issue again, i suggest replacing this w/ a retry loop that checks the logs every few seconds, giving up to, say, 30s for the expected log entry to appear.

that way we don't unnecessarily prolong the test in most cases, but also don't flake when things are slow.